lib/sensors/mux.cpp

```diff
@@ -53,7 +54,9 @@ core::Result Mux<T, N>::closeAllChannels()
 template<typename T, std::uint8_t N>
 std::optional<std::array<T, N>> Mux<T, N>::readAllChannels()
 {
-  std::array<T, N> mux_data;
+  // Zero-initialize the array
+  std::array<T, N> mux_data{};
+  std::uint8_t num_unusable_sensors = 0;
   for (std::uint8_t i = 0; i < N; ++i) {
     const auto sensor          = sensors_[i];
```

Comment:
I just noticed this: This is a copy. Make the lhs a reference.

Reply:
Also the RHS should use `.at(i)` for consistency.

Reply:
oh yes, this could have been really bad - thanks for pointing it out

Reply:
well, if it could have been really bad then you change your class to not allow this. for me it's a performance thing. if you don't want these sensors to be copied then you should go and remove the copy constructors.

Reply:
i tried doing this for II2cMuxSensors instances but that didn't work. now with just a reference to the array, without templating, the same code throws errors if you don't do `&sensor` but with templating doesn't. am a bit confused.

Reply:
`shared_ptr` on the other hand does seem to work

Reply:
templates aren't checked unless you put a concrete value in because no code is generated; this is why we need the dummy tests! (among other reasons)